lib/satellite/controller_extensions.rb

```diff
@@ -84,10 +94,33 @@ def skip_satellite_authentication?
       !!@skip_satellite_authentication
     end
 
+    def skip_satellite_authentication
+      @skip_satellite_authentication = true
+    end
+
+    def should_skip_satellite_authentication?
+      (params[:skip].to_i == 1).tap { |skip| delete_user_identity_cookie if skip }
```

Comment:
Looking at this method name, it seems like it could have some unintended side effects. I would typically expect a method with a question mark to simply return a boolean without any side effects, but it looks like this method takes a destructive action, and then returns a boolean.

Do you think it would make sense to possibly split this out into two functions, one that only returns the boolean like `should_skip_satellite_authentication?` and then one that actually uses the method, like `skip_satellite_authentication! if should_skip_satellite_authentication?` Just throwing out ideas.